clients/lsp-lua.el

```diff
@@ -556,6 +556,87 @@ and `../lib` ,exclude `../lib/temp`.
   :server-id 'lua-language-server
   :download-server-fn #'lsp-lua-language-server-install))
 
+;;; lua-language-server-latest
+
+(defgroup lsp-lua-language-server-latest nil
+  "Lua LSP client, provided by the Lua Language Server (GitHub release)."
+  :group 'lsp-mode
+  :link '(url-link "https://github.com/sumneko/lua-language-server"))
+
+(defcustom lsp-clients-lua-language-server-latest-bin
+  (f-join lsp-clients-lua-language-server-install-dir
+          "bin/"
+          (pcase system-type
+            ('gnu/linux "lua-language-server")
+            ('darwin "lua-language-server")
+            ('windows-nt "lua-language-server.exe")
+            (_ "lua-language-server")))
+  "Location of Lua Language Server (latest)."
+  :group 'lsp-lua-language-server-latest
+  :risky t
+  :type 'file)
+
+(defcustom lsp-clients-lua-language-server-latest-main-location
+  (f-join lsp-clients-lua-language-server-install-dir
+          "main.lua")
+  "Location of Lua Language Server main.lua (latest)."
+  :group 'lsp-lua-language-server-latest
+  :risky t
+  :type 'file)
+
+(defcustom lsp-clients-lua-language-server-latest-args '("-E")
+  "Arguments to run the Lua Language server (latest)."
+  :group 'lsp-lua-language-server-latest
+  :risky t
+  :type '(repeat string))
+
+(defun lsp--find-latest-lua-ls-release-url ()
+  "Fetch the latest version of lua-language-server from its github repo
+(for use with `lsp-lua-language-server-install'). Search for \"linux-x64\",
+\"darwin-x64\" and \"win32-x64\" tags on releases."
+  (let ((system-suffix (pcase system-type
+                         ('gnu/linux "linux-x64")
+                         ('darwin "darwin-x64")
+                         ('windows-nt "win32-x64")
+                         (_ "linux-x64")))
+        (url-request-method "GET"))
+    (with-current-buffer (url-retrieve-synchronously "https://api.github.com/repos/sumneko/lua-language-server/releases/latest")
+      ;; Go to beginning of the last line (where the retrieved json with information
+      ;; is in the buffer from `url-retrieve-synchronously')
+      (goto-char (point-max))
+      (goto-char (point-at-bol))
+      (let* ((json-result (prog1 (json-read) (kill-buffer)))
+             (os-search-pred (lambda (json-entry)
+                      (string-match-p system-suffix (alist-get 'name json-entry)))))
+        (message "Latest version found: %s" (alist-get 'name json-result))
+        (alist-get 'browser_download_url (seq-find os-search-pred (alist-get 'assets json-result)))
```

Comment:
nit: please close on the same line.

Reply:
you mean to make the (os-search-pred ...) section of the `let` in the same line, yes?

Reply:
closing brackets should be on the same line.